AlwaysOn/Wearable/src/main/java/com/example/android/wearable/wear/alwayson/MainActivity.kt

```diff
@@ -0,0 +1,357 @@
+
+        /** If the display is low-bit in ambient mode. i.e. it requires anti-aliased fonts.  */
+        private var isLowBitAmbient = false
+
+        /**
+         * If the display requires burn-in protection in ambient mode, rendered pixels need to be
+         * intermittently offset to avoid screen burn-in.
+         */
+        private var doBurnInProtection = false
+
+        /** Prepares the UI for ambient mode.  */
+        override fun onEnterAmbient(ambientDetails: Bundle) {
+            super.onEnterAmbient(ambientDetails)
+            isLowBitAmbient = ambientDetails.getBoolean(AmbientModeSupport.EXTRA_LOWBIT_AMBIENT, false)
+            doBurnInProtection =
+                ambientDetails.getBoolean(AmbientModeSupport.EXTRA_BURN_IN_PROTECTION, false)
+
+            // Cancel any active updates
+            activeUpdateJob.cancel()
+
+            /*
+             * Following best practices outlined in WatchFaces API (keeping most pixels black,
+             * avoiding large blocks of white pixels, using only black and white, and disabling
+             * anti-aliasing, etc.)
+             */
+            binding.state.setTextColor(Color.WHITE)
+            binding.updateRate.setTextColor(Color.WHITE)
+            binding.drawCount.setTextColor(Color.WHITE)
+            if (isLowBitAmbient) {
+                binding.time.paint.isAntiAlias = false
+                binding.timeStamp.paint.isAntiAlias = false
+                binding.state.paint.isAntiAlias = false
+                binding.updateRate.paint.isAntiAlias = false
+                binding.drawCount.paint.isAntiAlias = false
+            }
+            refreshDisplayAndSetNextUpdate()
+        }
+
+        /**
+         * Updates the display in ambient mode on the standard interval. Since we're using a custom
+         * refresh cycle, this method does NOT update the data in the display. Rather, this method
+         * simply updates the positioning of the data in the screen to avoid burn-in, if the display
+         * requires it.
+         */
+        override fun onUpdateAmbient() {
+            super.onUpdateAmbient()
+
+            /*
+             * If the screen requires burn-in protection, views must be shifted around periodically
+             * in ambient mode. To ensure that content isn't shifted off the screen, avoid placing
+             * content within 10 pixels of the edge of the screen.
+             *
+             * Since we're potentially applying negative padding, we have ensured
+             * that the containing view is sufficiently padded (see res/layout/activity_main.xml).
+             *
+             * Activities should also avoid solid white areas to prevent pixel burn-in. Both of
+             * these requirements only apply in ambient mode, and only when this property is set
+             * to true.
+             */
+            if (doBurnInProtection) {
+                binding.container.updatePadding(
```

Comment:
In my experience, it's weird to set negative padding values. What do you think about setting `translationX` and/or `translationY` instead?

Reply:
Agreed, updated in latest commit.